jvector-base/src/main/java/io/github/jbellis/jvector/util/ThreadPerPhysicalCorePool.java

```diff
@@ -0,0 +1,25 @@
+package io.github.jbellis.jvector.util;
+
+import java.util.concurrent.ForkJoinPool;
+import java.util.function.Supplier;
+
+public class ThreadPerPhysicalCorePool {
+    private static final int physicalCoreCount = Integer.getInteger("jvector.physical_core_count", Math.max(1, Runtime.getRuntime().availableProcessors()/2));
```

Comment:
this is a hard problem now :(

my intel cpu has 8P+8E cores, only the P are HT.  so the correct answer for my box is 16 but this gives 12.

additionally, some (upcoming? already existing?) server parts ONLY have single-threaded E cores.  AMD is moving this direction too.

Reply:
That's why there's a override :) a startup script could check the physical cores

Reply:
Deserves a mention in README?